packages/api/cms-api/src/page-tree/site-preview.resolver.ts

```diff
@@ -1,56 +1,32 @@
 import { Inject } from "@nestjs/common";
-import { Args, ArgsType, Field, ObjectType, Query, Resolver } from "@nestjs/graphql";
-import { IsNumber, IsString } from "class-validator";
-import { createHmac } from "crypto";
-import { differenceInMinutes, getTime } from "date-fns";
+import { Args, ArgsType, Field, Query, Resolver } from "@nestjs/graphql";
+import { IsObject, IsString } from "class-validator";
+import { GraphQLJSONObject } from "graphql-type-json";
+import jsonwebtoken from "jsonwebtoken";
 
 import { SITE_PREVIEW_CONFIG } from "./page-tree.constants";
 
-@ObjectType()
 @ArgsType()
-class SitePreviewHash {
-    @Field(() => Number)
-    @IsNumber()
-    timestamp: number;
-
+class SitePreviewArgs {
     @Field(() => String)
     @IsString()
-    hash: string;
+    path: string;
+
+    @Field(() => GraphQLJSONObject)
+    @IsObject()
+    previewData: JSON;
```

Comment:
Why is this a JSON and not strongly typed? For extendability in applications?

Reply:
Now it is typed, however, in cms-site the type has to be copied.